src/utils/wallet.js

```diff
@@ -438,53 +647,266 @@ class Wallet {
+        const recoveryKeysED = recoveryMethods.data.map((rm) => rm.publicKey)
+        // 3. get recovery keys that are NOT seed phrases && NOT null publicKey
+        const fak2lak = recoveryMethods.data.filter(({ kind, publicKey }) => kind !== 'phrase' && publicKey !== null).map((rm) => toPK(rm.publicKey))
+        // 4. push localStorage keys that are NOT recovery keys
+        fak2lak.push(...accountKeys.filter((ak) => !recoveryKeysED.includes(ak.public_key)).map((ak) => toPK(ak.public_key)))
+        // these are the keys we need to convert to limited access keys
+        console.log('fak2lak', fak2lak)
+        // 5. get the server public key for this accountId (confirmOnlyKey)
+        const getAccessKey = await this.postSignedJson('/2fa/getAccessKey', {
+            accountId
+        })
+        if (!getAccessKey || !getAccessKey.success || !getAccessKey.publicKey) {
+            console.log('error getting publicKey from contract-helper')
+            return
         }
+        const confirmOnlyKey = toPK(getAccessKey.publicKey)
+        console.log('confirmOnlyKey', confirmOnlyKey)
+
+        // const confirm = window.confirm('deploy contract?')
+        // if (!confirm) return
+        /********************************
+        Key updates, multisig deployment and initialization 
+        ********************************/
+        const newArgs = new Uint8Array(new TextEncoder().encode(JSON.stringify({ 'num_confirmations': 2 })));
+        const actions = [
+            // delete FAKs and add them as LAKs
+            ...fak2lak.map((pk) => nearApiJs.transactions.deleteKey(pk)),
+            ...fak2lak.map((pk) => nearApiJs.transactions.addKey(pk, nearApiJs.transactions.functionCallAccessKey(accountId, METHOD_NAMES_LAK, null))),
+            // confirmOnlyKey
+            nearApiJs.transactions.addKey(confirmOnlyKey, nearApiJs.transactions.functionCallAccessKey(accountId, METHOD_NAMES_CONFIRM, null)),
+            // deploy and initialize contract
+            nearApiJs.transactions.deployContract(contractBytes),
+            nearApiJs.transactions.functionCall('new', newArgs, 10000000000000, '0'),
+        ]
+        console.log('actions', actions)
+        return await account.signAndSendTransaction(accountId, actions);
+    }
 
+    async createNewAccountForTempAccount(accountId, fundingContract, fundingKey) {
+        // create account because recovery is validated
+        if (!accountId) {
+            // updated to get tempAccount.accountId
+            accountId = this.getAccountId()
+        }
+        await this.createNewAccount(accountId, fundingContract, fundingKey)
+        const account = this.getAccount(accountId)
+        console.log('account created', account)
+        // remove the temp account, we now have a real account on chain
+        delTempAccount()
+        return account
+    }
+
+    async setupRecoveryMessage(accountId, method, securityCode, fundingContract, fundingKey) {
```

Comment:
not sure why it has to be so complicated now

Reply:
cleaned up and majority moved to twoFactor.js